src/veh_interact.cpp

```diff
@@ -1729,6 +1733,10 @@ void veh_interact::display_details(const vpart_info &part)
 
     wborder(w_details, LINE_XOXO, LINE_XOXO, LINE_OXOX, LINE_OXOX, LINE_OXXO, LINE_OOXX, LINE_XXOO, LINE_XOOX);
 
+    if ( part == NULL ) {
+        wrefresh(w_details);
+        return;
+    }
```

Comment:
Just a note for the future: to avoid having to change all the `part.` to `part->` you can rename the parameter to `part_` or similar, and define a local variable `auto &part = *part_;` (after checking that the pointer is not null).

Now the remaining code can use `part.` (which has still the same meaning) like before and you don't need to change it to `part->`. Instead of dereferencing the pointer before calling the unction, it's now done inside the function.